lib/api/bucketGet.js

```diff
@@ -29,6 +31,177 @@ import { errors } from 'arsenal';
+            '</Owner>',
+            `<StorageClass>${v.StorageClass}</StorageClass>`,
+            '</Contents>'
+        );
+    });
+    list.CommonPrefixes.forEach(item => {
+        xml.push(
+            `<CommonPrefixes><Prefix>${item}</Prefix></CommonPrefixes>`
+        );
+    });
+    xml.push('</ListBucketResult>');
+    return xml.join('');
+}
+
+// Sample XML response for versioning:
+// http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGETVersion.html
+/*
+<?xml version="1.0" encoding="UTF-8"?>
+
+<ListVersionsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01">
+    <Name>bucket</Name>
+    <Prefix>my</Prefix>
+    <KeyMarker/>
+    <VersionIdMarker/>
+    <MaxKeys>5</MaxKeys>
+    <IsTruncated>false</IsTruncated>
+    <Version>
+        <Key>my-image.jpg</Key>
+        <VersionId>3/L4kqtJl40Nr8X8gdRQBpUMLUo</VersionId>
+        <IsLatest>true</IsLatest>
+        <LastModified>2009-10-12T17:50:30.000Z</LastModified>
+        <ETag>&quot;fba9dede5f27731c9771645a39863328&quot;</ETag>
+        <Size>434234</Size>
+        <StorageClass>STANDARD</StorageClass>
+        <Owner>
+            <ID>canonical-user-id</ID>
+            <DisplayName>mtd@amazon.com</DisplayName>
+        </Owner>
+    </Version>
+    <DeleteMarker>
+        <Key>my-image.jpg</Key>
+        <VersionId>03jpff543dhffds434rfdsFDN943fdsFkdmqnh892</VersionId>
+        <IsLatest>true</IsLatest>
+        <LastModified>2009-11-12T17:50:30.000Z</LastModified>
+        <Owner>
+            <ID>canonical-user-id</ID>
+            <DisplayName>mtd@amazon.com</DisplayName>
+        </Owner>
+    </DeleteMarker>
+</ListVersionsResult>
+*/
+
+function formatVersionsXML(bucketName, listParams, encoding, list) {
+    const xml = [];
+    xml.push(
+        '<?xml version="1.0" encoding="UTF-8"?>',
+        '<ListVersionsResult xmlns="http://s3.amazonaws.com/doc/' +
+            '2006-03-01/">',
+        `<Name>${bucketName}</Name>`
+    );
+
+    const xmlParams = [
+        { tag: 'Prefix', value: listParams.prefix },
+        { tag: 'MaxKeys', value: listParams.maxKeys },
+        { tag: 'KeyMarker', value: listParams.keyMarker },
+        { tag: 'VersionIdMarker', value: listParams.versionIdMarker },
+    ];
+
+    xmlParams.forEach(param => {
+        if (param.value) {
+            xml.push(`<${param.tag}>${param.value}</${param.tag}>`);
+        } else {
+            xml.push(`<${param.tag}/>`);
+        }
+    });
+
+    list.Contents.forEach(item => {
+        const key = encoding === 'url' ?
+            querystring.escape(item.key) : escapeForXML(item.key);
+        const v = VSUtils.decodeVersion(item.value);
+        const size = v['content-length'];
+        const ETag = v['content-md5'];
+        const lastModified = v['last-modified'];
+        const owner = {
+            DisplayName: v['owner-display-name'],
+            ID: v['owner-id'],
+        };
+        const storageClass = v['x-amz-storage-class'];
+
+        const versionId = VSUtils.getts(v);
```

Comment:
If you are using getters on some attributes might as well be consistent and do it for all.